pkgs/development/python-modules/dirsearch/default.nix

```diff
@@ -0,0 +1,86 @@
+{
+  lib,
+  fetchFromGitHub,
+  buildPythonPackage,
+  python,
+  # deps
+  beautifulsoup4,
+  certifi,
+  cffi,
+  chardet,
+  charset-normalizer,
+  colorama,
+  cryptography,
+  defusedxml,
+  idna,
+  jinja2,
+  markupsafe,
+  pyopenssl,
+  pyparsing,
+  pysocks,
+  pyspnego,
+  requests,
+  requests-ntlm,
+  setuptools,
+  urllib3,
+}:
+
+buildPythonPackage rec {
+  pname = "dirsearch";
+  version = "0.4.3";
+
+  src = fetchFromGitHub {
+    owner = "maurosoria";
+    repo = "dirsearch";
+    rev = "v${version}";
+    hash = "sha256-eXB103qUB3m7V/9hlq2xv3Y3bIz89/pGJsbPZQ+AZXs=";
+  };
+
+  # setup.py does some weird stuff with mktemp
+  postPatch = ''
+    substituteInPlace setup.py \
+      --replace-fail 'os.chdir(env_dir)' "" \
+      --replace-fail 'shutil.copytree(os.path.abspath(os.getcwd()), os.path.join(env_dir, "dirsearch"))' ""
+  '';
+
+  dependencies = [
+    beautifulsoup4
+    certifi
+    cffi
+    chardet
+    charset-normalizer
+    colorama
+    cryptography
+    defusedxml
+    idna
+    jinja2
+    markupsafe
+    pyopenssl
+    pyparsing
+    pysocks
+    pyspnego
+    requests
+    requests-ntlm
+    setuptools
```

Comment:
This should probably be in `build-system` instead, as we don't want to propagate this.

Reply:
moving `setuptools` to `build-system` makes `dirsearch` error with this message:
```
Traceback (most recent call last):
  File "/nix/store/i9hhlzdvhcdw6kyngl82amy5l1nx2m6w-python3.12-dirsearch-0.4.3/bin/.dirsearch-wrapped", line 6, in <module>
    from dirsearch.dirsearch import main
  File "/nix/store/i9hhlzdvhcdw6kyngl82amy5l1nx2m6w-python3.12-dirsearch-0.4.3/lib/python3.12/site-packages/dirsearch/dirsearch.py", line 23, in <module>
    from pkg_resources import DistributionNotFound, VersionConflict
ModuleNotFoundError: No module named 'pkg_resources'
```
[according to stackoverflow](https://stackoverflow.com/questions/7446187/no-module-named-pkg-resources) `pkg_resources` is part of `setuptools`